### ISSUE TITLE
iD 2.19.6 damaged imageries for Poland

### ISSUE THREAD

**Issue by maro-21** [openstreetmap/iD#8383]
The version 2.19.6 updated only imageries, but it broke down imageries for Poland:
* Instead of default background with the newest aerial photos, there is a black background now.
* On the top of the list there is an old imagery from 2014 which was marked as "default". This will make people draw nonexistent buildings from old photos. From my experience, browsing new users' edits, I can see that people don't change their default imagery. In case they want to change - it is not signed which one is newer. We have imagery from 2020 and this one should be default.
* The most important imagery which we draw buildings from, polska.e-mapa.net, was removed for no reason.

Could you restore it to the previous state?
We have a lot of active users in Poland which map every day. This version made difficult to map.

**Comment by manfredbrandl** [comment]
There is indeed 2.19.6 on openstreetmap.org since 3/3 but CHANGELOG shows 2.19.5 as latest release.

**Comment by mmd-osm** [comment]
You're probably looking at the one in the ~master~ develop branch? Try this one instead: https://github.com/openstreetmap/iD/blob/release/CHANGELOG.md

To roll back to a previous version, you need to comment on https://github.com/openstreetmap/openstreetmap-website/pull/3118 instead. Even better would be fixing the underlying issue.

**Comment by manfredbrandl** [comment]
I was looking at branch develop as i assumed the latest changes to appear there. Was that wrong? BTW there is no master branch visible for me.

**Comment by ttomasz** [comment]
this is a duplicate of #8382 (at least the main issue from point 1)

**Comment by Asteliks** [comment]
> The version 2.19.6 updated only imageries, but it broke down imageries for Poland:
> 
> * Instead of default background with the newest aerial photos, there is a black background now.
> * On the top of the list there is an old imagery from 2014 which was marked as "default". This will make people draw nonexistent buildings from old photos. From my experience, browsing new users' edits, I can see that people don't change their default imagery. In case they want to change - it is not signed which one is newer. We have imagery from 2020 and this one should be default.
> * The most important imagery which we draw buildings from, polska.e-mapa.net, was removed for no reason.
> 
> Could you restore it to the previous state?
> We have a lot of active users in Poland which map every day. This version made difficult to map.

First dot: seems to be related with: https://github.com/openstreetmap/iD/issues/8382 -> The problem seems to be that iD chooses the alphabetically first layer that has the “"best": true” for the region. In this case the alphabetically first layer is the overlay called “Geoportal 2 Nazwy ulic” thus the layers are not chosen, and the background is black because the overlay with street names is chosen. As a fix I can recommend not considering overlays for displayed backgrounds when entering the iD editor. 

Second dot: the layers are alphabetically sorted. Thus the layer "Geoportal 2: High Resolution Orthophotomap (aerial image)" is first on the list. If we could set the layer "Geoportal 2: Orthophotomap (aerial image)" as default we would have. As a side note the layer "Geoportal 2: High Resolution Orthophotomap (aerial image)" has the ""best": true" attribute because it is from GUGiK (an official trusted source for Poland) and thus can be trusted.

Third dot: https://github.com/osmlab/editor-layer-index/issues/1078

**Comment by matkoniecz** [comment]
> Second dot: the layers are alphabetically sorted. Thus the layer "Geoportal 2: High Resolution Orthophotomap (aerial image)" is first on the list. If we could set the layer "Geoportal 2: Orthophotomap (aerial image)" as default we would have. As a side note the layer "Geoportal 2: High Resolution Orthophotomap (aerial image)" has the ""best": true" attribute because it is from GUGiK (an official trusted source for Poland) and thus can be trusted.

Is either of this two from 2014 and outdated? Outdated imagery from official source should not get `best` parameter.

**Comment by Asteliks** [comment]
> > Second dot: the layers are alphabetically sorted. Thus the layer "Geoportal 2: High Resolution Orthophotomap (aerial image)" is first on the list. If we could set the layer "Geoportal 2: Orthophotomap (aerial image)" as default we would have. As a side note the layer "Geoportal 2: High Resolution Orthophotomap (aerial image)" has the ""best": true" attribute because it is from GUGiK (an official trusted source for Poland) and thus can be trusted.
> 
> Is either of this two from 2014 and outdated? Outdated imagery from official source should not get `best` parameter.

This is a hard question because the way GUGiK works is they add the ortophoto maps sent to them the by the local units all over Poland as soon as these local units have made new maps for the region. This is more of a national map data aggregator thus there can be sources with older images, but this is not explicitly stated. Basically, when there are several maps for one region the standard server (“Geoportal 2: Orthophotomap (aerial image)”) sorts the maps collected by the creation date and the high-resolution (“Geoportal 2: High Resolution Orthophotomap (aerial image)”) sorts them by the resolution.

TLDR.: “Geoportal 2: High Resolution Orthophotomap (aerial image)” can have older maps but they are also the latest maps available in higher quality.

**Comment by matkoniecz** [comment]
Thanks for an explanation! I finally understand how it works. I created https://github.com/osmlab/editor-layer-index/issues/1083 to better communicate it in general.

**Comment by maro-21** [comment]
The issue no longer occurs so I'm closing it.
